shell/browser/api/electron_api_web_frame_main.cc

```diff
@@ -141,17 +142,21 @@ v8::Local<v8::Promise> WebFrameMain::ExecuteJavaScript(
     return handle;
   }
 
-  if (user_gesture) {
-    auto* ftn = content::FrameTreeNode::From(render_frame_);
-    ftn->UpdateUserActivationState(
-        blink::mojom::UserActivationUpdateType::kNotifyActivation,
-        blink::mojom::UserActivationNotificationType::kTest);
-  }
-
-  render_frame_->ExecuteJavaScriptForTests(
-      code, base::BindOnce([](gin_helper::Promise<base::Value> promise,
-                              base::Value value) { promise.Resolve(value); },
-                           std::move(promise)));
+  static_cast<content::RenderFrameHostImpl*>(render_frame_)
+      ->ExecuteJavaScriptForTests(
+          code, user_gesture, true /* resolve_promises */,
+          content::ISOLATED_WORLD_ID_GLOBAL,
+          base::BindOnce(
+              [](gin_helper::Promise<base::Value> promise,
+                 blink::mojom::JavaScriptExecutionResultType type,
+                 base::Value value) {
+                if (type ==
+                    blink::mojom::JavaScriptExecutionResultType::kSuccess)
+                  promise.Resolve(value);
+                else
+                  promise.Resolve(base::Value());
```

Comment:
Can we reject the promise in the case of an error? Then we can match the behavior of WebContents.executeJavaScript.

Reply:
we should also add a test for this case too